src/librepgp/stream-common.cpp

```diff
@@ -910,9 +925,22 @@ init_tmpfile_dest(pgp_dest_t *dst, const char *path, bool overwrite)
         RNP_LOG("failed to build file path");
         return RNP_ERROR_BAD_PARAMETERS;
     }
+#ifdef HAVE_MKSTEMP
+    int fd = mkstemp(tmp);
+    if (fd < 0) {
+        RNP_LOG("failed to create temporary file with tempate '%s'. Error %d.", tmp, errno);
+        return RNP_ERROR_WRITE;
+    }
+#else
     mktemp(tmp);
```

Comment:
Is `mktemp()` implementation secure enough in this case (i.e. under Windows) ? Did you consider to use something specific for Windows, like `GetTempFileNameA()`

Reply:
@ni4 Yes, I have done some development this way. It entails massive complications, including splitting path and file name (using Windows SDK), temp file name becomes exactly 12 chars long, so the concept of renaming the temp file just by casting off the tail falls apart.  It's easier to implement `mkstemp` directly like this: https://stackoverflow.com/questions/6036227/mkstemp-implementation-for-win32

Reply:
> It's easier to implement mkstemp directly like this:

Agree, it would be easier to have our own implementation then. Maybe as a separate issue.

Reply:
@ni4 I have implemented `rnp_mkstemp` (well, copied from GNU source code) for Windows MSVC only -- turns out that `mkstemp` is available on all other platforms -- no errors.

Reply:
@rrrooommmaaa Thanks. Yeah, MinGW has it's own implementation. What about moving code to the file-utils.cpp instead of the separate file, for less clutter? Other then this (and license question) LGTM.

Reply:
@ni4 Moved the function `rnp_mkstemp` to `file-utils`. It builds ok, the only failure is transient `homebrew` error, right?
Please review.
@ronaldtse we have GNU Lesser General Public License mentioned in `rnp_mkstemp` implementation in `file-utils.cpp`
Is it ok to commit like this?

Reply:
@rrrooommmaaa LGTM, thanks! Restarted the macOS build, it happens with homebrew periodically. Feel free to merge once there is a reply from @ronaldtse

Reply:
...and after the second approval of course, believed there is one already.

Reply:
@rrrooommmaaa the LGPL license is not compatible with our BSD license, which is well answered in this [SO post](https://opensource.stackexchange.com/questions/6698/can-i-use-code-from-an-lgpl-v2-licenced-project-in-a-bsd-3-clause-licenced-proje)

> You cannot use LGPL code in a BSD-licensed project. When you are using LGPL'ed code the LGPL behaves very similar to the GPL. You can then only publish your project under the LGPL or GPL (even though some individual files in your project that do not depend on the LGPLed parts may use compatible licenses such as BSD).

> A LGPL-licensed library can be used by a BSD-licensed project, as long as users (incl. all end users) are free to study and swap out the LGPLed parts of the software. This means either
> * dynamically linking to the LGPL library, or
> distributing your project in a way that allows it to be relinked with a different version.

> Note that the LGPL doesn't just bind you but also other people distributing your project. This would go beyond the usual requirements of the BSD license, so it is worth documenting these additional obligations in your project.

Could you re-write this portion or somehow seek alternative functions under the MSVC environment? Sorry about this but thanks...

Reply:
Or @rrrooommmaaa is it possible to utilize the original functions/macros without including it in our code base?

Reply:
Ping @rrrooommmaaa ?

Reply:
I started to edit it, yes

Reply:
As Windows filesystem is case-insensitive, it needed some changes anyway

Reply:
This version has very littler of original code left, is it good enough?

Reply:
I think the new implementation can be considered an original implementation. Thank you @rrrooommmaaa !!